api/log.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	chimiddleware "github.com/go-chi/chi/middleware"
+	gcontext "github.com/netlify/gocommerce/context"
+)
+
+func newStructuredLogger(logger *logrus.Logger) func(next http.Handler) http.Handler {
+	return chimiddleware.RequestLogger(&structuredLogger{logger})
+}
+
+type structuredLogger struct {
+	Logger *logrus.Logger
+}
+
+func (l *structuredLogger) NewLogEntry(r *http.Request) chimiddleware.LogEntry {
+	entry := &structuredLoggerEntry{Logger: logrus.NewEntry(l.Logger)}
+	logFields := logrus.Fields{}
+
+	logFields["component"] = "api"
+	logFields["method"] = r.Method
+	logFields["path"] = r.URL.Path
+
+	if reqID := gcontext.GetRequestID(r.Context()); reqID != "" {
```

Comment:
I think that you have [this final message](https://github.com/netlify/gocommerce/pull/68/files/a36722f5cce20305af91f1b6a032bad228ed7a5a#diff-e527dd4db3e74ad6c53c84156b69fbe4R63) and that should be fine.